### DISCUSSION TITLE
how to add a permission to add messages to a user?

### DISCUSSION THREAD

**Discussion by mike-ivc-inform** [frePPLe/frepple#521] [Q&A]
Hello!
(win10, 6.14.0, Pgs12)
when I assign role groups, I cannot assign a role to add and change a comment. Therefore, a user who has group access rights assigned cannot create messages in the message tab.
![img0171](https://user-images.githubusercontent.com/87300645/161993671-0aaafbce-d755-44bb-a118-b44b938bd485.png)

To give permission to add messages to groups, I commented out the line in the code, `class Comment(models.Model):`

![img0172](https://user-images.githubusercontent.com/87300645/161993995-d745d907-bacd-48bb-9715-e015fdf9a5cf.png)

after that I was able to give permissions to groups to add and modify messages.
![img0173](https://user-images.githubusercontent.com/87300645/161994986-b459a192-54c1-4069-9ade-61519b1aecf4.png)

QUESTION: I think this is the wrong decision. What are the options without interfering with the kernel code?

**Comment by jdetaeye** [comment]
Your suggestion will work.

In my opninion, it's not the best the solution:  
The comment permission you created allows the user to comment on any object.   
A better design would be to allow anybody with the "change" permission to enter comments - Eg "change item" permission allows you to leave comments on items, but not on sales orders, resources, ...

In any case, there is some code cleanup to do to get the permissions right for the comments...  Creating an issue from this discussion.
